### PR TITLE
#162 Support Test Recording

### PR DESCRIPTION
**Pull request by oomelianchuk**

### CODE REVIEW

**Review by wurzelkuchen — CHANGES_REQUESTED** [review]
Part one of my review. More will follow.

File: src/test/java/com/xceptance/neodymium/tests/visual/ai/Test_Images_ESF.network
Comment (wurzelkuchen) [inline src/test/java/com/xceptance/neodymium/tests/visual/ai/Test_Images_ESF.network:1]: This seems out of place, video recording does not use visual compare. Check if this is used somewhere in the code, if not, remove it from this branch.

**Review by wurzelkuchen — CHANGES_REQUESTED** [review]

File: src/main/java/com/xceptance/neodymium/recording/config/GifRecordingConfigurations.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.xceptance.neodymium.recording.config;
+
+import org.aeonbits.owner.Config.Sources;
+
+@Sources(
+{
+  "${recording.temporaryConfigFile}", "system:env", "system:properties", "file:config/dev-gif-recording.properties", "file:config/gif-recording.properties"
```
Comment (wurzelkuchen) [inline src/main/java/com/xceptance/neodymium/recording/config/GifRecordingConfigurations.java:7]: We should add neodymium.properties and dev-neodymium properties here, to get a better coverage, with less confusion, which property needs to be where

File: src/main/java/com/xceptance/neodymium/recording/config/RecordingConfigurations.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Configuration interface, which contains on the values needed for both gif and video recording. Needed to enable
+ * abstract handling of the subclasses. Contains the key and default values for gif as this format is default for
+ * recording
+ * 
+ * @author olha
+ */
+@LoadPolicy(LoadType.MERGE)
+public interface RecordingConfigurations extends Mutable
```
[the remark this answers is not in the document]
Comment (kqmpetenz) [reply src/main/java/com/xceptance/neodymium/recording/config/RecordingConfigurations.java:15]: Do you want one RecordingConfiguration Class combining the two (video and gif) or do you want a specific configuration class for each format?

**Review by wurzelkuchen — APPROVED** [review]

**Pull request merged**